etl/steps/data/garden/gcp/2022-09-29/global_carbon_budget_additional.py

```diff
@@ -0,0 +1,269 @@
+"""This step creates a dataset that has additional variables that are currently not included in the Global Carbon
+Budget (GCB) dataset (which was created in importers).
+
+In the future (next time GCB dataset is updated and moved to ETL), a newer version of this step should create the
+entire GCB dataset.
+
+"""
+
+from typing import Tuple, cast
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from owid.datautils import geo
+from shared import CURRENT_DIR
+
+from etl.helpers import Names
+
+# Regions and income groups to create (by aggregating), following OWID definitions.
+REGIONS = [
+    "Africa",
+    "Asia",
+    "Europe",
+    "European Union (27)",
+    "North America",
+    "Oceania",
+    "South America",
+    "Low-income countries",
+    "Upper-middle-income countries",
+    "Lower-middle-income countries",
+    "High-income countries",
+]
+# Variables to aggregate and type of aggregation to apply.
+AGGREGATES = {"production_emissions": "sum", "consumption_emissions": "sum"}
+
+# Naming conventions.
+N = Names(str(CURRENT_DIR / "global_carbon_budget_additional"))
```

Comment:
There is a N.directory option in case you wanted to have:

`N = Names(__file__)`
`current_dir = N.directory`

would mean your `shared.py` is no longer needed, but it's just a different way of doing the same thing!

Reply:
Here I'm using the `shared.py` file simply to avoid having to deal with `__file__` when working with jupyter notebooks. I have never found a better workaround (apart from typing in a new cell `__file__ = "/Users/..."` which is quite annoying). If you know any other, please let me know!

Reply:
Ah okay! I'm not aware of any alternatives and I didn't realise it was annoying to anyone except me 😄